pkg/deploy/controller/deployment/factory.go

```diff
@@ -122,6 +126,15 @@ func (factory *DeploymentControllerFactory) Create() controller.RunnableControll
 	}
 }
 
+// setPodAttributes updates the deployer pod attributes based on the deployment strategy configuration
+func (factory *DeploymentControllerFactory) setPodAttributes(strategy *deployapi.DeploymentStrategy, pod *kapi.Pod) {
+	switch strategy.Type {
+	case deployapi.DeploymentStrategyTypeCustom:
+		util.MergeInto(pod.Labels, strategy.Labels, 0)
```

Comment:
Got it, hadn't checked the MergeInto impl. Add a comment here noting that we're not overwriting existing labels on the pod?